workers/src/data/extractors.rs

```diff
@@ -0,0 +1,170 @@
+//! Extract data from encoded payloads.
```

Comment:
So guys, i found out why nothing is decoding for you with alloy, you're modelling the structs wrong! :joy:  This is a situation where if something isn't working you can't bruteforce. It's telling you it doesnt work and it's showing you in the fixture.

There are bugs here to do with packet offsets.

Reply:
https://github.com/LayerZero-Labs/LayerZero-v2/blob/7bcfb4d5dac4192570af5e51dbc67413a6116a14/packages/layerzero-v2/evm/protocol/contracts/interfaces/ILayerZeroEndpointV2.sol

https://github.com/LayerZero-Labs/LayerZero-v2/blob/7bcfb4d5dac4192570af5e51dbc67413a6116a14/packages/layerzero-v2/evm/messagelib/contracts/uln/LzExecutor.sol#L16

https://github.com/LayerZero-Labs/LayerZero-v2/blob/7bcfb4d5dac4192570af5e51dbc67413a6116a14/packages/layerzero-v2/solana/programs/libs/oapp/src/lib.rs#L11

Reply:
Still, `endpoint.lzReceive` will have these parameters

https://github.com/LayerZero-Labs/LayerZero-v2/blob/7bcfb4d5dac4192570af5e51dbc67413a6116a14/packages/layerzero-v2/evm/messagelib/contracts/uln/LzExecutor.sol#L121

Reply:
That doesnt change anything i said